python/lsst/analysis/tools/atools/wholeSkyPlotTool.py

```diff
@@ -0,0 +1,90 @@
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ("WholeSkyPlotTool",)
+
+from lsst.pex.config import Field, ListField
+
+from ..actions.plot.wholeSkyPlot import WholeSkyPlot
+from ..interfaces import AnalysisTool
+
+
+class WholeSkyPlotTool(AnalysisTool):
+    """Plot metrics across all tracts on the sky."""
+
+    bands = ListField[str](
+        doc="Photometric bands to use for plots.",
+        default=["u", "g", "r", "i", "z", "y"],
+        listCheck=lambda x: len(set(x)) == len(x),
+    )
+    parameterizedBand = Field[bool](
+        doc="Does this AnalysisTool support band as a name parameter", default=False
+    )
+    plotKeys = ListField[str](
+        doc="Metrics to plot that are not band-specific.",
+        default=["yPerpPSF_yPerp_psfFlux_median"],
+        listCheck=lambda x: len(set(x)) == len(x),
+    )
+    keysWithBand = ListField[str](
+        doc="Metrics to plot that are band-specific.",
+        default=["e1Diff_{band}_highSNStars_median"],
+        itemCheck=lambda x: "{band}" in x,
+        listCheck=lambda x: len(set(x)) == len(x),
+    )
+    xAxisLabel = Field[str](doc="Label for the x axis.", default="RA (degrees)")
+    yAxisLabel = Field[str](doc="Label for the y axis.", default="Dec (degrees)")
+    xLimits = ListField[float](doc="Plotting limits for the x axis.", default=[-5.0, 365.0])
+    yLimits = ListField[float](doc="Plotting limits for the y axis.", default=[-10.0, 60.0])
+    figureSize = ListField[float](doc="Size of the figure.", default=[9.0, 3.5])
+
+    propagateData = True
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.prep.keysToLoad = [
+            "corners",
+            "tract",
+        ]
+        self.produce.plot = WholeSkyPlot()
+
+    def finalize(self):
+        # Loop over bands to explicitly define names of band-specific metrics.
+        for key in self.keysWithBand:
+            self.plotKeys.extend([key.format(band=band) for band in self.bands])
+        # This should be in validate, but for some reason finalize gets called
+        # after.
+        if len(set(self.plotKeys)) != len(self.plotKeys):
+            raise ValueError(f"{self.plotKeys=} has duplicate keys.")
```

Comment:
This section here with the ValueErrors is the only thing I might want to change... I'm not sure why validate is running after finalize.